include/seqan3/alignment/pairwise/policy/banded_score_trace_dp_matrix_policy.hpp

```diff
@@ -0,0 +1,289 @@
+        auto zip_score = ranges::view::zip(std::span{std::addressof(*current_matrix_iter), span},
+                                           std::span{std::addressof(*(current_matrix_iter + 1)), span});
+        return ranges::view::zip(std::move(zip_score),
+                                 ranges::view::iota(col_begin, col_end),
+                                 std::span{std::addressof(*trace_matrix_iter), span});
+    }
+
+    //!\brief Moves internal matrix pointer to the next column.
+    constexpr void next_column() noexcept
+    {
+        base_t::next_column();
+        // Move trace back pointer to begin of next column
+        trace_matrix_iter = std::ranges::begin(trace_matrix) + band_size * current_column_index;
+        // As long as we shift the band towards the band_column_index, jump to the correct begin.
+        if (current_column_index < band_column_index)
+            trace_matrix_iter += band_column_index - current_column_index;
+    }
+
+    /*!\brief Parses the traceback starting from the given coordinate.
+     * \param end_coordinate The coordinate from where to start the traceback.
+     *
+     * \returns A tuple containing the begin coordinate and a tuple with all seqan3::detail::gap_segment s for the
+     *          first sequence and the second sequence.
+     */
+    constexpr auto parse_traceback(alignment_coordinate const & end_coordinate) const
+    {
+        // Store the trace segments.
+        std::deque<gap_segment> first_segments{};
+        std::deque<gap_segment> second_segments{};
+
+        // Put the iterator to the position where the traceback starts.
+        auto direction_iter = std::ranges::begin(trace_matrix);
+        std::ranges::advance(direction_iter, end_coordinate.first_seq_pos * band_size +
+                                     end_coordinate.second_seq_pos);
+
+        // Parse the trace until interrupt.
+        while (*direction_iter != trace_directions::none)
+        {
+            // parse until end of diagonal run
+            while (static_cast<bool>(*direction_iter & trace_directions::diagonal))
+            {
+                std::ranges::advance(direction_iter, -static_cast<int_fast32_t>(band_size));
+            }
+
+            size_t col_pos = std::ranges::distance(std::ranges::begin(trace_matrix), direction_iter) / band_size;
+            // parse vertical gap -> record gap in first_segments (will be translated into gap of first sequence)
+            if (static_cast<bool>(*direction_iter & trace_directions::up) ||
+                static_cast<bool>(*direction_iter & trace_directions::up_open))
+            {
+                // Get the current column index (note the column based layout)
+
+                gap_segment gap{col_pos, 0u};
+
+                // Follow gap until open signal is detected.
+                while (!static_cast<bool>(*direction_iter & trace_directions::up_open))
```

Comment:
Do you really need this cast? Souldn't this be contextually convertile to bool anyway?

Reply:
nop